common/client/send_only_node.go

```diff
@@ -0,0 +1,180 @@
+package client
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"sync"
+
+	"github.com/smartcontractkit/chainlink/v2/common/types"
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+	"github.com/smartcontractkit/chainlink/v2/core/utils"
+)
+
+type SendOnlyClient[
```

Comment:
I think for now we can unexport this interface so it doesn't pollute the API of the package. In the future though, once we decide on a testing strategy for this package, it will make more sense to have a targeted client API that can be easily mocked instead of having an entire RPC interface.